Assets/Scripts/InteractLogic.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Interfaces;
+using Player;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+
+[RequireComponent(typeof(IMovable))]
+public class InteractLogic : MonoBehaviour
+{
+    /// <summary>
+    /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+    /// (Won't conflict with character attacking enemy or moving on the map)
+    /// </summary>
+    /// 
+
+    [Tooltip("Showing the object player is focusing")]
+    [SerializeField]
+    private Interactable focus;
+
+    [Tooltip("Showing the object target player is interacting")]
+    [SerializeField]
+    private Transform target;
+
+    private IMovable _movable;
+
+    private NavMeshAgent _navMeshAgent;
+
+    private void Awake()
+    {
+        _movable = GetComponent<IMovable>();
+        _navMeshAgent = GetComponent<NavMeshAgent>();
```

Comment:
You could think about giving Movable a public method SetStoppingDistance. In my opinion it's better to do it all via the Movable instead of getting the NavMeshAgent here.

Reply:
is done !
```suggestion
        public void SetStoppingDistance(float distance)
        {
            NavMeshAgent.stoppingDistance = distance;
        }
```

Reply:
```suggestion
        public void SetStoppingDistance(float distance)
        {
            NavMeshAgent.stoppingDistance = distance;
        }
```

Reply:
is done !
```suggestion
        public void SetStoppingDistance(float distance)
        {
            NavMeshAgent.stoppingDistance = distance;
        }
```